src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/EdgeCasesTests.cs

```diff
@@ -23,6 +23,99 @@ public static partial class EdgeCasesTests
+            //  PFX private key off disk depending on how the original was created.
+            //
+            //    - If the original was created using the X509Certificate2 constructor, disposing the Find-created clone will
+            //      not delete the on-disk key.
+            //
+            //    - If the original was created by importing it into an X509Certificate2Collection, disposing the Find-created
+            //      clone *will* delete the on-disk key. (and when you dispose of the original, it will again try to
+            //      delete that key and silently eat the failure. Unless of course, someone in the meantime persisted 
+            //      a new key with the same name, in which case, he gets bollixed.)
+            //
+            //  This is due to the fact that the Import() api thumbtacks a "please delete key on dispose" note onto the native CERT_CONTEXT 
+            //  object by setting a framework-invented "property" (CERT_DELETE_KEYSET_PROP_ID) on it. This property
+            //  carries over to any X509Certificate2 instance that wraps this handle (either by the Find() api
+            //  or by the X509Certificate2 constructor that takes an IntPtr handle.
+            //
+            //  In contrast, the X509Certificate2 constructor that takes a PFX does not set this pseudo-property.
+            //
+            //  This inconsistent X509 behavior has been around too long to change safely now. As far as EnvelopedCms is concerned,
+            //  the important test is to make sure it doesn't use that Find() api, or in any way internally clone
+            //  any "extraStore" certificates you pass in.
+            //
+
+            byte[] encodedMessage =
+                ("3082010c06092a864886f70d010703a081fe3081fb0201003181c83081c5020100302e301a311830160603550403130f5253"
+                + "414b65795472616e7366657231021031d935fb63e8cfab48a0bf7b397b67c0300d06092a864886f70d01010105000481805e"
+                + "bb2d08773594be9ec5d30c0707cf339f2b982a4f0797b74d520a0c973d668a9a6ad9d28066ef36e5b5620fef67f4d79ee50c"
+                + "25eb999f0c656548347d5676ac4b779f8fce2b87e6388fbe483bb0fcf78ab1f1ff29169600401fded7b2803a0bf96cc160c4"
+                + "96726216e986869eed578bda652855c85604a056201538ee56b6c4302b06092a864886f70d010701301406082a864886f70d"
+                + "030704083adadf63cd297a86800835edc437e31d0b70").HexToByteArray();
+
+            EnvelopedCms ecms = new EnvelopedCms();
+            ecms.Decode(encodedMessage);
+
+            using (X509Certificate2 cert = Certificates.RSAKeyTransfer1.LoadPfxUsingCollectionImport())
+            {
+                X509Certificate2Collection extraStore = new X509Certificate2Collection(cert);
+                ecms.Decrypt(extraStore);
+
+                byte[] expectedContent = { 1, 2, 3 };
+                ContentInfo contentInfo = ecms.ContentInfo;
+                Assert.Equal<byte>(expectedContent, contentInfo.Content);
+            }
```

Comment:
I don't understand what part of this test would fail under the described conditions.  If it's that decrypt will throw that the cert got eaten, can you add a comment at that line?